jobbergate-slurp/slurp/s3_ops.py

```diff
@@ -1,140 +1,181 @@
 """
 Provides a convenience class for managing calls to S3.
 """
+import asyncio
 import contextlib
+import itertools
+import re
 import tarfile
 import tempfile
-from collections import namedtuple
 from io import BytesIO
 from pathlib import Path
+from typing import Set
 
-from fastapi import UploadFile
-from file_storehouse import FileManager, FileManagerReadOnly, client
-from file_storehouse.engine.s3 import EngineS3
-from file_storehouse.key_mapping import KeyMappingNumeratedFolder
-from jobbergate_api.s3_manager import LIST_OF_TRANSFORMATIONS, ApplicationFileManager
+import aioboto3
+from buzz import check_expressions, handle_errors, require_condition
 from loguru import logger
 
 from slurp.config import settings
 
+APPLICATIONS_WORK_DIR = "applications"
+APPLICATION_CONFIG_FILE_NAME = "jobbergate.yaml"
+APPLICATION_SOURCE_FILE_NAME = "jobbergate.py"
+APPLICATION_TEMPLATE_FOLDER = "templates"
 
-def build_managers():
-
-    db_gen = namedtuple("db_generation", "legacy nextgen")
-    s3_folder = namedtuple("Folder", "applications jobscripts")
-
-    s3_client = db_gen(
-        legacy=client(
-            "s3",
-            endpoint_url=settings.LEGACY_S3_ENDPOINT_URL,
-            aws_access_key_id=settings.LEGACY_AWS_ACCESS_KEY_ID,
-            aws_secret_access_key=settings.LEGACY_AWS_SECRET_ACCESS_KEY,
-        ),
-        nextgen=client(
-            "s3",
-            endpoint_url=settings.NEXTGEN_S3_ENDPOINT_URL,
-            aws_access_key_id=settings.NEXTGEN_AWS_ACCESS_KEY_ID,
-            aws_secret_access_key=settings.NEXTGEN_AWS_SECRET_ACCESS_KEY,
-        ),
-    )
 
-    s3man = s3_folder(
-        applications=db_gen(
-            legacy=FileManagerReadOnly(
-                engine=EngineS3(
-                    s3_client.legacy,
-                    settings.LEGACY_S3_BUCKET_NAME,
-                    "jobbergate-resources",
-                ),
-            ),
-            nextgen=ApplicationFileManager(
-                bucket_name=settings.NEXTGEN_S3_BUCKET_NAME,
-                s3_client=s3_client.nextgen,
-            ),
-        ),
-        jobscripts=db_gen(
-            legacy=None,
-            nextgen=FileManager(
-                engine=EngineS3(s3_client.nextgen, settings.NEXTGEN_S3_BUCKET_NAME, "job-scripts"),
-                transformation_list=LIST_OF_TRANSFORMATIONS,
-                key_mapping=KeyMappingNumeratedFolder("jobbergate.txt"),
-            ),
-        ),
-    )
+@contextlib.asynccontextmanager
+async def s3_bucket(is_legacy: bool = True):
+    """
+    Context manager used to get a bucket from S3.
+
+    It is connected to the legacy bucket when is_legacy is True, and to the nextgen bucket otherwise.
```

Comment:
```suggestion
    It is connected to the legacy bucket when is_legacy is True, otherwise,  it is connected to the nextgen bucket .
```

Reply:
solved in [ec1a607](https://github.com/omnivector-solutions/jobbergate/pull/181/commits/ec1a6078ac96b075160525f22303b719aebc9425)